GH-10083 Add Nullability to the mapping package

cppwfs opened the pull request:
Related to: #10083
* AbstractHeaderMapper.java - Enhanced null safety by adding @Nullable annotations to fields and methods.
* Updated method signatures to match those in AbstractHeaderMapper for the following classes:
   - DefaultAmqpHeaderMapper.java
   - DefaultSoapHeaderMapper.java
   - DefaultXmppHeaderMapper.java

<!--
Thanks for contributing to Spring Integration. 
Please provide a brief description of your pull-request and reference any related issue numbers (prefix references with #) or StackOverflow questions.

See the [Contributor Guidelines for more information](https://github.com/spring-projects/spring-integration/blob/main/CONTRIBUTING.adoc).
-->

cppwfs committed:
GH-10083 Add Nullability to the mapping package

Related to: #10083
* AbstractHeaderMapper.java - Enhanced null safety by adding @Nullable annotations to fields and methods.
* Updated method signatures to match those in AbstractHeaderMapper for the following classes:
   - DefaultAmqpHeaderMapper.java
   - DefaultSoapHeaderMapper.java
   - DefaultXmppHeaderMapper.java

artembilan commented on spring-integration-core/src/main/java/org/springframework/integration/mapping/AbstractHeaderMapper.java:
```diff
@@ -308,7 +310,9 @@ private ResolvableType createJsonResolvableTypHeaderInAny(Object typeId, @Nullab
 		try {
-			return JsonHeaders.buildResolvableType(getClassLoader(), typeId, contentId, keyId);
+			var classLoader = getClassLoader();
+			Assert.state(classLoader != null, "No ClassLoader available");
```
Looks like Spring deals with null `ClassLoader` everywhere.
We should revise that `JsonHeaders.buildResolvableType()` to accept `null`, respectively.
In the end it uses `ClassUtils.forName()` which accepts null well.

artembilan commented on spring-integration-core/src/main/java/org/springframework/integration/mapping/support/JsonHeaders.java:
```diff
@@ -103,9 +105,9 @@ else if (classValue != null) {
+	public static ResolvableType buildResolvableType(@Nullable Class<?> targetClass, @Nullable Class<?> contentClass,
 			@Nullable Class<?> keyClass) {
-
+		Assert.state(targetClass != null, "The targetClass must not be null");
```
This is not correct.
The `targetClass` must not be null.

Consider to mark `getClassForValue()` with a `@Contract` we talked yesterday in other place:
```
@Contract("_, null -> null; _, !null -> !null")
```

cppwfs committed:
buildResolvableType should accept a null ClassLoader

Add Contract for buildResolvableType where the first param can be anything, and if the 2 or 3 value is not null, it will return a not null.

artembilan commented on spring-integration-core/src/main/java/org/springframework/integration/mapping/support/JsonHeaders.java:
```diff
@@ -105,9 +106,9 @@ else if (classValue != null) {
 	 * @since 5.2.4
 	 */
-	public static ResolvableType buildResolvableType(@Nullable Class<?> targetClass, @Nullable Class<?> contentClass,
+	@Contract("_, !null, _ -> !null; _, _, !null -> !null")
```
This is wrong.
The method does not return `null` at all.
See `getClassForValue()`.
That's where the `@Contract` has to be.

artembilan commented on spring-integration-core/src/main/java/org/springframework/integration/mapping/support/JsonHeaders.java:
```diff
@@ -69,18 +70,18 @@ private JsonHeaders() {
 		Class<?> keyClass = getClassForValue(classLoader, keyClassValue);
 		Class<?> contentClass = getClassForValue(classLoader, contentClassValue);
-
+		Assert.notNull(targetClass, "targetClass must not be null");
```
When you fix `getClassForValue()` with the `@Contract`, this is not necessary.

artembilan commented on spring-integration-core/src/main/java/org/springframework/integration/mapping/support/JsonHeaders.java:
```diff
@@ -69,18 +70,18 @@ private JsonHeaders() {
 		return buildResolvableType(targetClass, contentClass, keyClass);
 	}
 
 	@Nullable
```
Please, move this next to the return type.

cppwfs committed:
Move Contract to getClassForValue

Remove associated Assert.state.

artembilan reviewed (approved):
LGMT.
Will merge when build is green.
